File: ruby/json-se.rb
require "json"

obj = {
  foo1: "bar",
  foo2: 2,
  foo3: [1, 2, 3],
  foo4: nil,
}
repeat_count = 10000000

start = Time.now
repeat_count.times { JSON.dump(obj) }
_end = Time.now

puts ((_end - start) * 1000).to_i

File: perl/json-se.pl
use Time::HiRes qw/gettimeofday tv_interval/;
use JSON::XS qw/encode_json/;

my $obj = +{
  foo1 => "bar",
  foo2 => 2,
  foo3 => [1, 2, 3],
  foo4 => undef,
};
my $repeat_count = 10000000;

my $start = [gettimeofday];
for (my $i = 0; $i < $repeat_count; ++$i) {
  encode_json $obj;
}
my $end = [gettimeofday];
print int(tv_interval($start, $end) * 1000), "\n";
